lib/wibox/widget/template.lua

```diff
@@ -362,34 +362,28 @@ end
 -- @method clone
 -- @treturn wibox.widget.template The copy.
 function template:clone()
-    return template {
-        template = self._private.widget_template
-    }
+    return template(self._private.widget_template)
 end
 
 --- Create a new `wibox.widget.template` instance.
--- @tparam[opt] table args
--- @tparam[opt] table|widget args.template The widget template to use.
--- @tparam[opt] function args.update_callback The callback function to update
+-- @tparam[opt] table tmpl
+-- @tparam[opt] function tmpl.update_callback The callback function to update
 --   the widget.
--- @tparam[opt] boolean args.update_now Update the widget after its
+-- @tparam[opt] boolean tmpl.update_now Update the widget after its
 --   construction. This will call the `:update()` method with no parameter.
 -- @treturn wibox.widget.template The new instance.
 -- @constructorfct wibox.widget.template
-function template.new(args)
-    args = args or {}
+function template.new(tmpl)
+    tmpl = tmpl or {}
 
     local ret = wbase.make_widget(nil, nil, { enable_properties = true })
 
     gtable.crush(ret, template, true)
     ret._private.connections = {}
 
-    ret:set_template(args.template)
-    ret:set_update_callback(args.update_callback)
-    ret:set_update_now(args.update_now)
-
-    -- Apply the received buttons, visible, forced_width and so on
-    gtable.crush(ret, args)
+    ret:set_template(tmpl)
+    ret:set_update_callback(tmpl.update_callback)
+    ret:set_update_now(tmpl.update_now)
```

Comment:
At this point, I think a separate `create_callback` would make more sense than using this boolean to treat `update_callback` as if it was a `create_callback`.
And this would be required for the "drop-in replacement".

Reply:
For the drop-in part, I was just planning at leaving the old code in place. The issue is the function signature. It has the tag/client as part of it, so it's hard to mutualize into a common API. There are ways around it, but it seems simpler to omit `create_callback` given it's not very useful. Something like `gears.connection` would be a better fit. I may extract it from the `gears.reactive` PR (even if they share some code) since it isn't as affected by the linting issues as the other 2 parts.

The whole callback thing is quite unique in our API when you think about it. We nearly always use signals for this kind of stuff. As you pointed elsewhere, it *does* make sense to keep it here because it makes the users life easier (I agree). `gears.connection` would overlap, but would still have some learning curve issue and has some footguns (like connecting to signals which don't exist).